src/App.jsx

```diff
@@ -1,17 +1,39 @@
-// import { useState } from 'react'
+import { useEffect, useState } from 'react'
 import './App.css'
 import TodoList from './TodoList.jsx';
 import AddTodoForm from './AddTodoForm.jsx';
 
+function useSemiPersistentState() {
+  const TODO_LIST_KEY = 'savedTodoList';
```

Comment:
careful with using SCREAMING_SNAKE_CASE - in JS, this format is usually reserved for environmental variables to TypeScript enums so that they stick out from regular code. No need to change but I wanted you to be aware that this could cause some initial confusion for someone onboarding to this codebase.